scim/client.bal

```diff
@@ -0,0 +1,262 @@
+import ballerina/oauth2;
+import ballerina/http;
+import ballerina/io;
+
+# This client is used to connect to the SCIM2 APIs of Asgardeo for Inbound User Provisioning.
+@display {label: "SCIM Client Connector"}
+public class Client {
+    private final string orgName;
+    private final string clientId;
+    private final string clientSecret;
+    private final string[] scope;
+    final oauth2:ClientOAuth2Provider provider;
+    final http:Client clientEndpoint;
+    final map<string> headers;
+
+    # Initializes the SCIM client.
+    #
+    # + orgName - The name of the organization
+    # + clientId - The client ID of the application
+    # + clientSecret - The client secret of the application
+    # + scope - permitted scopes
+    public isolated function init(string orgName, string clientId, string clientSecret, string[] scope) returns error? {
+        self.orgName = orgName;
+        self.clientId = clientId;
+        self.clientSecret = clientSecret;
+        self.scope = scope;
+
+        self.provider = new ({
```

Comment:
Is there a specific reason for using an imperative approach here? The code could potentially be simplified by using the declarative approach outlined in the [specification](https://ballerina.io/spec/oauth2/#521-declarative-approach-http-client)

Reply:
Check following BBE as well:
https://ballerina.io/learn/by-example/http-client-oauth2-client-credentials-grant-type/